### PR TITLE
bind to and test IPv6 when the interface family exists

### LINKED ISSUE #186: fix ipv6 test on Travis
we discovered flakiness around the ipv6 test on Travis in #182. 

Let's get to the bottom of it, since it's likely that's just a travis config thing

### PR DESCRIPTION
**Pull request by reqshark**
fixes #186 the travis IPv6 test

let's test and bind to an IPv6 family address when such an interface exists

also updated our travis box to use trusty so hopefully builds start faster

@nickdesaulniers, review?

### CODE REVIEW

File: test/sockoptapi.js [hunk 1]
```diff
@@ -94,6 +94,21 @@ test('sockopt api methods', function(t){
...
+    for (var iface in interfaces) {
+      var ipv6 = interfaces[iface].filter(function fam(i){
+        return i.family === 'IPv6';
+      });
+      if (ipv6.length) {
+        break;
+      }
+    }
+
+    if (!ipv6.length) {
```
Comment (nickdesaulniers) [inline test/sockoptapi.js]: I know `ipv6` will get hoisted, but do you mind declaring it earlier & outside of the for loop?

**Comment by nickdesaulniers** [comment]
LGTM, just that nit on the hoisting.

**Comment by reqshark** [comment]
@nickdesaulniers ok good point, updated

**Comment by nickdesaulniers** [comment]
Thanks buddy! :wolf: :kick_scooter:

**Pull request merged**